enum.c

```diff
@@ -2984,13 +2984,12 @@ enum_member(VALUE obj, VALUE val)
 }
 
 static VALUE
-each_with_index_i(RB_BLOCK_CALL_FUNC_ARGLIST(i, memo))
+each_with_index_i(RB_BLOCK_CALL_FUNC_ARGLIST(_, index))
 {
-    struct MEMO *m = MEMO_CAST(memo);
-    VALUE n = imemo_count_value(m);
+    struct vm_ifunc *ifunc = rb_current_ifunc();
+    ifunc->data = (const void *)rb_int_succ(index);
```

Comment:
Would you mind adding a comment explaining what the ifunc does, what is its purpose etc?

This code can be fairly opaque to those not as familiar with CRuby internals.

Reply:
[c018336](https://github.com/ruby/ruby/pull/11868/commits/c018336e794403406820cd6af09e3728b972f873)
Sure! I think this kind of commentary sits better on the struct, though.